app/controllers/organizations_controller.rb

```diff
@@ -123,7 +123,7 @@ def search
   def authorize_organization_addition
     new_github_organization = github_organization_from_params
 
-    return if new_github_organization.admin?(current_user.github_user.login)
+    return if new_github_organization.admin?(current_user.github_user.login(use_cache: false))
```

Comment:
What would you think about extracting the login to a `current_user_login` method? I'm mostly wondering about memoizing this so we aren't busting the cache multiple times...

Reply:
@jeffrafter Sorry for the late :eyes:, could you elaborate here? The login will be memoized to `current_user.github_login` in this case.

Reply:
Sorry, I don't see where that is happening and might be confused. Looking at the code it looked like there were multiple code-paths in this controller that led to: `current_user.github_user.login(use_cache: false)`. This is the before filter, but lower it is hit in `create_user_organization_access` and that method is being called N times (where N is the number of classrooms here: https://github.com/education/classroom/blob/049a9c0ef89a5ae42f9bfb94a1c9b79bd1704edb/app/controllers/organizations_controller.rb#L173 (I think 🤐). My main concern here is that by adding `use_cache: false` we will be busting the cache every time the `login` is accessed in this controller which could be multiple. If that is true this controller would be using up to date information but would dramatically increase the potential number of API calls. 

That said; if this is being memoized somewhere I am missing then all of this is 💯

Reply:
This is fixed in https://github.com/education/classroom/pull/2198/commits/e76b055e347d66bbc479eeb43e72137099c2bdbf